src/foam/u2/layout/Layouts.js

```diff
@@ -1,23 +1,207 @@
+            .end();
+        }
+      })
+      return this;
+    },
+
+    function initE() {
+      this.SUPER();
     }
   ]
 });
 
 foam.CLASS({
   package: 'foam.u2.layout',
-  name: 'ColumnLayout',
+  name: 'Row',
   extends: 'foam.u2.Element',
+
+  documentation: `
+    An individual column row within a group of rows.
+  `,
+
+  methods: [
+    function initE() {
+      this.style({ 'display': 'block' }) // more specificity should enforce ruling
+    }
+  ],
+});
+
+foam.CLASS({
+  package: 'foam.u2.layout',
+  name: 'Cols',
+  extends: 'foam.u2.Element',
+
+  documentation: `
+    A layout for a group of columns layed out horizontally.
+    A row or div can contain numerous Cols elements
+  `,
+
+  requires: [
+    'foam.u2.layout.Col',
+    'foam.u2.layout.AlignmentTypes',
+  ],
+
+  css: `
+    ^ {
+      display: flex;
+    }
+  `,
+
+  properties: [
+    {
+      class: 'Map',
+      name: 'defaultConfig'
+    },
+    {
+      class: 'foam.u2.ViewSpec',
+      name: 'border',
+      value: { class: 'foam.u2.borders.NullBorder' }
+    },
+    {
+      class: 'Enum',
+      of: 'foam.u2.layout.AlignmentTypes',
+      name: 'horizontalAlignmentTypes',
+      documentation: `
+        An enum on horizontal alignment types
+        (i.e. on the web, this follows the 'justify-content' CSS property)
+      `,
+      value: null,
+    },
+    {
+      class: 'String',
+      name: 'horizontalAlignmentValue',
+      documentation: `
+        To render the proper CSS for web based on the alignment types enum.
+        Will later be used to also account for mobile.
+
+        NOTE: Added support for now to hedge for the future, but we will
+        use the direct flex columning as shown in foam.comics.v2.Comics for now
+      `,
+      expression: function(horizontalAlignmentTypes$webFlexProp){
+        return horizontalAlignmentTypes$webFlexProp;
+      }
+    }
+  ],
+
   methods: [
     function add() {
-      console.log('Adding column!');
-      return this.SUPER.apply(this, arguments);
+
+      /**
+       * Here we are checking for two cases
+       * 1. If a Col element is added, then call the super on it
+       * 2. If another element is added, then wrap it in a Col element with default configuration
+       */
+      [...arguments].forEach(value => {
+        if ( this.Col.isInstance(value) ) {
+          this.SUPER(value);
+        }
+        else {
+          // conditionally add isAlignmentAdded to defaultConfig if horizontal alignment was specified
+          if (this.horizontalAlignmentTypes){
+            this.defaultConfig.isAlignmentActivated = true;
+          }
+
+          this
+            .start(this.Col, this.defaultConfig)
+              .start(this.border)
+                .add(value)
+              .end()
+            .end();
+        }
+      })
+      return this;
+    },
+
+    function initE() {
+      this.SUPER();
+
+      // if horizontal alignment was specified
+      if (this.horizontalAlignmentValue) {
```

Comment:
Let's give a default to the alignmentType and assume we always have one. Then we don't need to be making these checks and we apply the justify-content unconditionally.

Reply:
done